### ISSUE TITLE
False positive S1854

### ISSUE THREAD

**Issue by eduherminio** [SonarSource/sonar-dotnet#3780]
### Description

I don't fully understand what's causing the issue: possible related to ranges and/or the usage of `string.Length`?

### Repro steps

The code piece is something like this, unless I'm missing something:

```csharp

var str1 = Guid.NewGuid().ToString();

var isTrue = false;
var str2 = isTrue        // S1854: Useless assignment
    ? ".."
    : "";

var str3 = str1[str2.Length..];

```

[SonarCloud analysis](https://sonarcloud.io/project/issues?branch=single-file&id=DotnetCombine&issues=AXXyTZW5Ek-lKlJoFcEc&open=AXXyTZW5Ek-lKlJoFcEc)

### Expected behavior

No code smell detected

### Actual behavior

S1854 code smell detected

### Related information

* SonarCloud
* .NET 5, C# 9

**Comment by pavel-mikula-sonarsource** [comment]
Hi @eduherminio ,

Thank you for reporting this case. I can confirm it as a FP related to ranges. We have this issue already reported in #3348 so I'll close this one.
